File: frontend/src/services/conversation.ts
import api from './api';
import { Message } from './message';

export interface Conversation {
  id: number;
  user_id: number;
  robot_id?: number;
  title: string;
  created_at: string;
  updated_at: string;
  messages?: Message[];
}

export interface ConversationCreate {
  title?: string;
  robot_id?: number;
}

export const conversationService = {
  async getConversations(): Promise<Conversation[]> {
    const response = await api.get<Conversation[]>('/conversations');
    return response.data;
  },

  async getConversation(id: number): Promise<Conversation> {
    const response = await api.get<Conversation>(`/conversations/${id}`);
    return response.data;
  },

  async createConversation(data: ConversationCreate): Promise<Conversation> {
    const response = await api.post<Conversation>('/conversations', data);
    return response.data;
  },

  async deleteConversation(id: number): Promise<void> {
    await api.delete(`/conversations/${id}`);
  },

  async updateTitle(id: number, title: string): Promise<void> {
    await api.patch(`/conversations/${id}/title`, null, {
      params: { title },
    });
  },
};

File: backend/app/schemas/conversation.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional
from .message import MessageResponse


class ConversationCreate(BaseModel):
    title: Optional[str] = "新对话"
    robot_id: Optional[int] = None


class ConversationResponse(BaseModel):
    id: int
    user_id: int
    robot_id: Optional[int] = None
    title: str
    created_at: datetime
    updated_at: datetime
    messages: List[MessageResponse] = []
    
    class Config:
        from_attributes = True


class ConversationListResponse(BaseModel):
    id: int
    user_id: int
    robot_id: Optional[int] = None
    title: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

